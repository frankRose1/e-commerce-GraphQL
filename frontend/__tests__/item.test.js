import ItemComponent from '../components/Item';
import {shallow} from 'enzyme';
import toJSON from 'enzyme-to-json';

const fakeItem = {
  id: "12341541",
  title: 'A fake item',
  price: 5012,
  description: 'Is it real or is it fake',
  image: 'pizza.jpg',
  largeImage: 'largepizza.jpg',
}

describe('<Item />', () => {

  it('matches the snapshot', () => {
    const wrapper = shallow(<ItemComponent item={fakeItem} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('renders and displays title and pricetag properly', () => {
    const wrapper = shallow(<ItemComponent item={fakeItem} />);
    const PriceTag = wrapper.find('PriceTag');
    expect(PriceTag.children().text()).toBe('$50.12');
    expect(wrapper.find('Title a').text()).toBe(fakeItem.title);
  });

  it('renders the images properly', () => {
    const wrapper = shallow(<ItemComponent item={fakeItem} />);
    const img = wrapper.find('img');
    expect(img.props().src).toBe(fakeItem.image);
    expect(img.props().alt).toBe(fakeItem.title);
  });

  it('renders the buttons properly', () => {
    const wrapper = shallow(<ItemComponent item={fakeItem} />);
    const buttonList = wrapper.find('.buttonList');
    expect(buttonList.children()).toHaveLength(3);
    expect(buttonList.find('Link').exists()).toBe(true);
    expect(buttonList.find('AddToCart').exists()).toBe(true);
    expect(buttonList.find('DeleteItem').exists()).toBe(true);
  });

});