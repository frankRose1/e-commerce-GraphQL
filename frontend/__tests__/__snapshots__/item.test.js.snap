// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Item /> matches the snapshot 1`] = `
<ItemStyles__Item>
  <img
    alt="A fake item"
    src="pizza.jpg"
  />
  <Title>
    <Link
      href={
        Object {
          "pathname": "/item",
          "query": Object {
            "id": "12341541",
          },
        }
      }
    >
      <a>
        A fake item
      </a>
    </Link>
  </Title>
  <PriceTag>
    $50.12
  </PriceTag>
  <p>
    Is it real or is it fake
  </p>
  <div
    className="buttonList"
  >
    <Link
      href={
        Object {
          "pathname": "/update",
          "query": Object {
            "id": "12341541",
          },
        }
      }
    >
      <a>
        Edit
      </a>
    </Link>
    <AddToCart
      itemId="12341541"
    />
    <DeleteItem
      id="12341541"
    />
  </div>
</ItemStyles__Item>
`;
